projects/chaos/src/powertrain_heartbeat.c

```diff
@@ -0,0 +1,101 @@
+#include "powertrain_heartbeat.h"
```

Comment:
Can you add a super brief summary explaining this? It was a little hard to understand with all the callbacks.

Reply:
Done. (Implementation in `*.c` and general in `*.h`).